x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/enrich/EnrichPolicy.java

```diff
@@ -5,142 +5,93 @@
 import org.elasticsearch.common.xcontent.XContentBuilder;
-import org.elasticsearch.common.xcontent.XContentHelper;
 import org.elasticsearch.common.xcontent.XContentParser;
-import org.elasticsearch.common.xcontent.XContentType;
-
-import java.io.IOException;
-import java.util.List;
-import java.util.Map;
-import java.util.Objects;
 
-/**
- * Represents an enrich policy including its configuration.
- */
-public final class EnrichPolicy implements Writeable, ToXContentFragment {
+public class EnrichPolicy implements Writeable, ToXContentFragment {
 
     public static final String ENRICH_INDEX_NAME_BASE = ".enrich-";
 
-    public static final String EXACT_MATCH_TYPE = "exact_match";
-    public static final String[] SUPPORTED_POLICY_TYPES = new String[]{EXACT_MATCH_TYPE};
-
-    private static final ParseField TYPE = new ParseField("type");
-    private static final ParseField QUERY = new ParseField("query");
-    private static final ParseField INDICES = new ParseField("indices");
-    private static final ParseField ENRICH_KEY = new ParseField("enrich_key");
-    private static final ParseField ENRICH_VALUES = new ParseField("enrich_values");
+    private static final ParseField NAME = new ParseField("name");
+    private static final ParseField VERSION_CREATED = new ParseField("version_created");
+    private static final ParseField DEFINITION = new ParseField("definition");
 
-    @SuppressWarnings("unchecked")
     private static final ConstructingObjectParser<EnrichPolicy, Void> PARSER = new ConstructingObjectParser<>("policy",
         args -> new EnrichPolicy(
             (String) args[0],
-            (QuerySource) args[1],
-            (List<String>) args[2],
-            (String) args[3],
-            (List<String>) args[4]
+            (Version) args[1],
+            (EnrichPolicyDefinition) args[2]
         )
     );
 
     static {
-        declareParserOptions(PARSER);
-    }
-
-    private static void declareParserOptions(ConstructingObjectParser<?, ?> parser) {
-        parser.declareString(ConstructingObjectParser.constructorArg(), TYPE);
-        parser.declareObject(ConstructingObjectParser.optionalConstructorArg(), (p, c) -> {
-            XContentBuilder contentBuilder = XContentBuilder.builder(p.contentType().xContent());
-            contentBuilder.generator().copyCurrentStructure(p);
-            return new QuerySource(BytesReference.bytes(contentBuilder), contentBuilder.contentType());
-        }, QUERY);
-        parser.declareStringArray(ConstructingObjectParser.constructorArg(), INDICES);
-        parser.declareString(ConstructingObjectParser.constructorArg(), ENRICH_KEY);
-        parser.declareStringArray(ConstructingObjectParser.constructorArg(), ENRICH_VALUES);
+        PARSER.declareString(ConstructingObjectParser.constructorArg(), NAME);
+        PARSER.declareField(ConstructingObjectParser.constructorArg(), (p, c) -> Version.fromString(p.text()), VERSION_CREATED,
```

Comment:
Can newer ES versions fromString versions that are not explicitly defined. e.g. what happens if version 6.8.0 is the version serialized can this read it ?

Reply:
Yes - The `fromString()` method condenses the version string given down into the integral version id. If the version id isn't defined in the known list of versions, it is still created, but the version of lucene that it associates with the returned version is a guess. I don't foresee this being an issue, but I could be off base here.